mempool/src/pool.rs

```diff
@@ -810,36 +847,137 @@ impl GetMemoryUsage for SystemUsageEstimator {
     }
 }
 
+pub(crate) struct MempoolInterfaceHandle {
```

Comment:
Too many `pub(crate)`... which indicate a design issue. I've always said that `pub(crate)` is most of the time not a good sign. Generally speaking, you have a style of putting to many unrelated things in one file. You may wanna think how to structure this into modules that make sense, where `pub` alone can give you the guarantee that only the mempool will see the crates.

For example, in #495, you can see that the `BlockMaker` is only accessible to the crate internally, and I never had to use `pub(crate)`, because the `BlockMaker` is isolated in its own submodule, which is declared only at one level higher, that needs it.